app/controllers/pages_controller.rb

```diff
@@ -123,10 +112,20 @@ def checkin
         .order("articles.published_at" => :desc)
         .first
 
-    if daily_thread
-      redirect_to URI.parse(daily_thread.path).path
-    else
-      redirect_to "/notifications"
-    end
+    redirect_daily_thread_request(daily_thread)
+  end
+
+  private
+
+  def redirect_daily_thread_request(daily_thread)
+    path = relative_path(daily_thread&.path) || notifications_path
```

Comment:
nitpick: the issue here is daily thread might be nil because first returns false (there's also a check for a nil path in the utility method - not sure if we need both guards - is there ever an article without a path?) - is there a clear way to express this without a safe navigation operator? (something tingles somewhere inside me when I see a safe navigation operator, and I end up having to do more thinking and surrounding reading to find out what it's guarding against and why). 

I see (`daily_thread && relative_path(daily_thread.path) || notifications_path`) feels a little clunky. 

Is it possible passing just the tag name from the actions is the right abstraction?

call sites look like `redirect_daily_thread_request("welcome")` or `redirect_daily_thread_request("challenge")`...

implementation is something like this? 

```
def redirect_daily_thread_request(tag_name)
  relation = Article.admin_published_with(tag_name)
  redirect_to relative_path(relation.first.path) if relation.any?
  
  redirect_to notifications_path
end
```

I'm likely over thinking this.